riot/riot.py

```diff
@@ -61,9 +61,62 @@ def to_list(x: t.Union[_K, t.List[_K]]) -> t.List[_K]:
     return [x] if not isinstance(x, list) else x
 
 
+@dataclasses.dataclass(unsafe_hash=True)
+class Interpreter:
+    hint: dataclasses.InitVar[str]
+
+    def __post_init__(self, hint: t.Union[float, int, str]) -> None:
+        """Normalize the data."""
+        self.hint = str(hint)
+
+    def __eq__(self, other: object) -> bool:
```

Comment:
This should also be done with dataclass AFAICT, you can use `compare=True` or `False` to configure it.